tensorboard/components/tensor_widget/tensor-widget-impl.ts

```diff
@@ -526,18 +543,35 @@ export class TensorWidgetImpl implements TensorWidget {
           this.renderSelection();
         });
         valueDiv.addEventListener('mouseenter', () => {
-          const detailedValue = valueDiv.getAttribute(DETAILED_VALUE_ATTR_KEY);
-          if (!detailedValue) {
+          // The DETAILED_VALUE_ATTR_KEY attribute of the value div holds a
+          // "detailed" string representation of the value of the corresponding
+          // tensor element. It is detailed in the sense that it has as many
+          // decimal points as supported by JavaScript's string representation
+          // of numbers, in the case of float dtypes. The only exceptions
+          // are very long string elements, which we still truncate in order
+          // to avoid overtaxing the DOM.
+          const valueAttr = valueDiv.getAttribute(DETAILED_VALUE_ATTR_KEY);
+          if (!valueAttr) {
             return;
           }
+          let detailedValue: number | boolean | string = JSON.parse(valueAttr);
```

Comment:
Alternate option that would resolve the below comments differently: don't `JSON.parse`, and instead just propagate `detailedValue` as a string only, regardless of the dtype.  It's just for the tooltip anyway, right?

Reply:
Adopting the suggestion of just storing the tooltip text in the attribute, without JSON stringify or parse. Thanks.